doc/reference/file_system/index.rst

```diff
@@ -3,13 +3,50 @@
 File Systems
 ############
 
+Zephyr RTOS File system Switch (FS) allows applications to mount multiple
+file systems at different mount points (ex: "/fatfs" and "/nffs"). The
+mount point data structure contains all the necessary information required
+to instantiate, mount and operate on file system. The File system Switch
+decouple the applications from directly accessing individual file system
+specific API's or internal functions by introducing file system registration
+mechanisms.
 
+In Zephyr, Any File system implementation or library can be plugged into or
+pulled out through file system registration API's.
 
-.. comment
-   not documenting
-   .. doxygengroup:: file_system
-   .. doxygengroup:: file_system_storage
+.. code-block:: c
 
+        int fs_register(enum fs_type type, struct fs_file_system_t *fs);
+
+.. code-block:: c
+
+        int fs_unregister(enum fs_type type, struct fs_file_system_t *fs);
+
+Zephyr RTOS supports multiple instances of a file system by making use of
+mount point as the disk volume name which is used by the file system library
+while formatting or mounting a disk.
+
+A file system is declared as:
+
+.. code-block:: c
+
+	static struct fs_mount_t mp = {
+	.type = FS_FATFS,
+	.mnt_point = FATFS_MNTP,
+	.fs_data = &fat_fs,
+	};
+
+where
+
+- ``FS_FATFS`` is the file system type like FATFS or NFFS.
+- ``FATFS_MNTP`` is the mount point where file system will be mounted.
+- ``fat_fs`` is the file system data which will by fs_mount() API.
+
```

Comment:
I would add "known limitation paragraph": NFFS is limited to one instance by its internally limitation.